LayoutTestTests/LayoutTestCaseTests/LYTLayoutFailingTestSnapshotRecorderTests.m

```diff
@@ -0,0 +1,304 @@
+    BOOL isDirectory = NO;
+    BOOL fileExisits = [[NSFileManager defaultManager] fileExistsAtPath:directoryPath isDirectory:&isDirectory];
+    
+    XCTAssertTrue(fileExisits);
+    XCTAssertTrue(isDirectory);
+}
+
+- (void)testSaveImageOfCurrentViewSavesImageToDisk {
+    UIView *testView = [[UIView alloc] initWithFrame:(CGRect){0, 0, 100, 100}];
+    [self.recorder startNewLogForClass:self.class];
+    
+    [self.recorder saveImageOfView:testView withData:@{@"key" : @"value"} fromInvocation:self.invocation failureDescription:@""];
+    
+    NSString *dataHash = @"";
+#ifdef __LP64__
+    dataHash = @"11921695704359177406";
+#else
+    dataHash = @"1252068542";
+#endif
+    NSString *imageName = [NSString stringWithFormat:@"/Width-100.00_Height-100.00_Data-%@.png", dataHash];
+    NSString *imagePath = [[self pathForCurrentInvocation] stringByAppendingString:imageName];
+    XCTAssertTrue([[NSFileManager defaultManager] fileExistsAtPath:imagePath]);
+}
+
+- (void)testSaveImageOfCurrentViewWhenViewHasWidthOfZeroDoesNotSaveImage {
+    UIView *testView = [[UIView alloc] initWithFrame:(CGRect){0, 0, 0, 100}];
+    [self.recorder startNewLogForClass:self.class];
+    
+    [self.recorder saveImageOfView:testView withData:nil fromInvocation:self.invocation failureDescription:@""];
+    
+    NSArray *filelist = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:[self pathForCurrentInvocation] error:nil];
+    XCTAssertFalse(filelist.count == 1);
+}
+
+- (void)testSaveImageOfCurrentViewWhenViewHasHeightOfZeroDoesNotSaveImage {
+    UIView *testView = [[UIView alloc] initWithFrame:(CGRect){0, 0, 100, 0}];
+    [self.recorder startNewLogForClass:self.class];
+    
+    [self.recorder saveImageOfView:testView withData:nil fromInvocation:self.invocation failureDescription:@""];
+    
+    NSArray *filelist = [[NSFileManager defaultManager] contentsOfDirectoryAtPath:[self pathForCurrentInvocation] error:nil];
+    XCTAssertEqual(0, filelist.count);
+}
+
+- (void)testSaveImageOfCurrentViewWhenDataHasSpecialCharactersImageSavedWithExpectedName {
+    UIView *testView = [[UIView alloc] initWithFrame:(CGRect){0, 0, 100, 100}];
+    NSDictionary *data = @{@"key" : @"漢語 ♔ 🚂 ☎ Here are some more special characters ˆ™£‡‹·Ú‹˜ƒª•"};
+    [self.recorder startNewLogForClass:self.class];
+    
+    [self.recorder saveImageOfView:testView withData:data fromInvocation:self.invocation failureDescription:@""];
+    
+    NSString *dataHash = @"";
+#ifdef __LP64__
+    dataHash = @"7040999336831705875";
+#else
+    dataHash = @"2409456403";
+#endif
+    NSString*expectedImageName = [NSString stringWithFormat:@"/Width-100.00_Height-100.00_Data-%@.png", dataHash];
+    NSString *imagePath = [[self pathForCurrentInvocation] stringByAppendingString:expectedImageName];
+    XCTAssertTrue([[NSFileManager defaultManager] fileExistsAtPath:imagePath]);
+}
+
+- (void)testSaveImageOfCurrentViewWhenDataDescriptionIsOverTwoHunderedCharactersImageSavedWithExpectedName {
```

Comment:
This is a hang over for when the data description was being added to the file name as OS X has a max file name length of 250. I think it's still a valid test though to cover the edge case.